java/all/src/main/scala/org/typelevel/otel4s/java/OtelJava.scala

```diff
@@ -48,30 +56,26 @@ object OtelJava {
     * @return
     *   An effect of an [[org.typelevel.otel4s.Otel4s]] resource.
     */
-  def forAsync[F[_]: LiftIO: Async](jOtel: JOpenTelemetry): F[Otel4s[F]] =
-    IOLocal(Vault.empty)
-      .map { implicit ioLocal: IOLocal[Vault] =>
+  def forAsync[F[_]: LiftIO: Async](jOtel: JOpenTelemetry): F[OtelJava[F]] =
```

Comment:
Is there a way to use `OtelJava` with `Vault` as a context?

Reply:
no. unfortunately, `Vault` is fundamentally incompatible with the API of `Context` from the Java library, precluding interoperability. what we have currently is only a hack to hold onto a few stored values that we care about, but is potentially losing nearly all of the actual context

Reply:
also, `Vault` is not compliant with spec, as the otel spec requires that keys have a name for debugging purposes